File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        int x,y;
        for(String c :tokens){
            if(c.equals("+")||c.equals("-")||c.equals("*")||c.equals("/")){
                y=stack.pop();
                x=stack.pop();
                if(c.equals("+")){
                    stack.push(x+y);
                }
                else if(c.equals("-")){
                    stack.push(x-y);
                }
                else if(c.equals("*")){
                    stack.push(x*y);
                }
                else if(c.equals("/")){
                    stack.push(x/y);
                }
            }
                else{
                    stack.push(Integer.parseInt(c));
                }
            }
        return stack.pop();
    }
}

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        #run a loop to scan the arrays
        for token in tokens:
            # check for integers
            if token not in {"+","-","*","/"}:
                stack.append(int(token))
            else:
                a = stack.pop()
                b = stack.pop()

                if token == "+":
                    stack.append(b + a)
                elif token == "-":
                    stack.append(b - a)
                elif token == "*":
                    stack.append(b * a)
                else:
                    #division where it truncates towards 0
                    stack.append(int(b / a))
        #return TOS
        return stack[-1]
